src/OrchardCore.Modules/OrchardCore.Workflows/Controllers/WorkflowTypeController.cs

```diff
@@ -127,17 +118,16 @@ public async Task<IActionResult> Index(WorkflowTypeIndexOptions options, PagerPa
                 .Take(pager.PageSize)
                 .ListAsync();
 
-            // The existing session's connection is returned, don't dispose it
-            var connection = await _session.CreateConnectionAsync();
-
             var dialect = _session.Store.Configuration.SqlDialect;
             var sqlBuilder = dialect.CreateBuilder(_session.Store.Configuration.TablePrefix);
             sqlBuilder.Select();
             sqlBuilder.Distinct();
             sqlBuilder.Selector(nameof(WorkflowIndex), nameof(WorkflowIndex.WorkflowTypeId), _session.Store.Configuration.Schema);
             sqlBuilder.Table(nameof(WorkflowIndex), alias: null, _session.Store.Configuration.Schema);
 
-            var workflowTypeIdsWithInstances = await connection.QueryAsync<string>(sqlBuilder.ToSqlString());
+            // Use existing session connection. Do not use 'using' or dispose the connection.
+            var connection = await _session.CreateConnectionAsync();
+            var workflowTypeIdsWithInstances = (await connection.QueryAsync<string>(sqlBuilder.ToSqlString(), param: null, _session.CurrentTransaction)).ToList();
```

Comment:
```suggestion
            var connection = await _session.CreateConnectionAsync();
            var transaction = await _session.BeginTransactionAsync();
            var workflowTypeIdsWithInstances = (await connection.QueryAsync<string>(sqlBuilder.ToSqlString(), param: null, transaction).ToList();
```

Reply:
we don't need a transaction here. I think if there is already a created transaction in the session, we should pass it. the `_session.CreateConnectionAsync()` return current session

Reply:
@Piedone so you are also saying we should not need to create a transaction here too?

Reply:
Yes.

Reply:
okay. if you good with this PR, please approve it. I would like to test it out in the new package tomorrow.